tianshou/policy/base.py

```diff
@@ -563,3 +581,52 @@ def _nstep_return(
         returns = rew[now].reshape(bsz, 1) + gamma * returns
     target_q = target_q * gamma_buffer[gammas].reshape(bsz, 1) + returns
     return target_q.reshape(target_shape)
+
+
+# TODO: is there another method somewhere in tianshou to do this? Can _nstep_return be used?
+@jit
+def calculate_episode_returns(rewards: Sequence[float], gamma: float) -> np.ndarray:
+    """Calculate discounted returns from a rewards of a single episode.
+
+    :param rewards: rewards of a single episode
+    :param gamma: discount factor
+    :return: a numpy array of shape (len(rewards), )
+    """
+    len_episode = len(rewards)
+    discounted_returns = np.zeros(len_episode)
+    discounted_returns[-1] = rewards[-1]
+
+    for j in range(len_episode - 2, -1, -1):
+        discounted_returns[j] = rewards[j] + gamma * discounted_returns[j + 1]
+
+    return discounted_returns
+
+
+@jit
+def calculate_returns_from_buffer(
+    buffer: ReplayBuffer,
+    gamma: float,
+    done_field: str = "done",
+    reward_field: str = "rew",
+) -> np.ndarray:
+    """Calculate discounted returns from a replay buffer, taking into.
```

Comment:
Docstring broken, but I can fix it in a subsequent PR. This one should be merged already